eng/common/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -145,24 +151,29 @@ function GetMatrixConfigFromJson([String]$jsonConfig)
             $config.displayNamesLookup.Add($_.Name, $_.Value)
         }
     }
-    $config.include = $config.include | Where-Object { $null -ne $_ } | ForEach-Object {
-        $ordered = [ordered]@{}
-        $_.PSObject.Properties | ForEach-Object {
-            $ordered.Add($_.Name, $_.Value)
-        }
-        return $ordered
-    }
-    $config.exclude = $config.exclude | Where-Object { $null -ne $_ } | ForEach-Object {
-        $ordered = [ordered]@{}
-        $_.PSObject.Properties | ForEach-Object {
-            $ordered.Add($_.Name, $_.Value)
+    if ($null -ne $config.replace) {
+        $config.replace.PSObject.Properties | ForEach-Object {
+            $config.replaceLookup.Add($_.Name, $_.Value)
         }
-        return $ordered
     }
+    $config.include = $config.include | ForEach-Object { ConvertPsObjectToOrderedDictionary $_ }
+    $config.exclude = $config.exclude | ForEach-Object { ConvertPsObjectToOrderedDictionary $_ }
 
     return $config
 }
 
+function ConvertPsObjectToOrderedDictionary([PSCustomObject]$obj)
```

Comment:
nit: ```Convert-PsObjectToOrderedDictionary``` ... generally speaking I think you've not used the ```Verb-Noun``` convention when naming these functions. It's triggering my PowerShell OCD :)

Reply:
Haha, I've been generally avoiding the convention because for these general library functions as I think it would make the names very convoluted (since they aren't CLI-facing). However, it's kind of funny I accidentally used a supported verb but with the wrong syntax here :D